sdks/go/pkg/beam/io/textio/textio.go

```diff
@@ -93,11 +96,18 @@ func readFn(ctx context.Context, filename string, emit func(string)) error {
 	}
 	defer fd.Close()
 
-	scanner := bufio.NewScanner(fd)
-	for scanner.Scan() {
-		emit(scanner.Text())
+	rd := bufio.NewReader(fd)
+	for {
+		line, err := rd.ReadString('\n')
+		if err == io.EOF {
```

Comment:
The documentation for ReadString states:

> If ReadString encounters an error before finding a delimiter, it returns the data read before the error and the error itself (often io.EOF)

So it sounds like you'd want to emit the line here once before breaking, maybe with a quick check `if len(line) != 0`.

Reply:
ack. done.